tools/gdb/dump_insert.py

```diff
@@ -0,0 +1,77 @@
+import gdb, sys
+ 
+# This script walks an insert list in WiredTiger, it does so by traversing the bottom level of the
+# skip list and printing the next pointer array. It dumps the data to "dump.txt" as the content
+# is often too large for the terminal.
+#
+# It takes a WT_INSERT_HEAD structure and a key format. In order to use it in gdb it must first be
+# sourced. Example usage:
+# source dump_insert.py
+# dump_insert WT_INSERT_HEAD,S
+#
+# It supports 3 key formats: u, i, S.
+class insert():
+    def __init__(self, key, address, next_array):
+        self.key = key
+        self.address = address
+        self.next_array = next_array
+ 
+    def print(self, output_file):
+        output_file.write("Key: " + str(self.key) + " Address: " + self.address + " Next array: " + str(self.next_array) + "\n")
+ 
+class dump_insert(gdb.Command):
+    key_format = 'S'
+    inserts = {}
+    def __init__(self):
+        super(dump_insert, self).__init__("dump_insert", gdb.COMMAND_DATA)
+
+    def usage(self):
+        print("usage:")
+        print("dump_insert WT_INSERT_HEAD,key_format")
+        exit(1)
+
+    def decode_key(self, key):
+        if self.key_format == 'S':
+            return(key.decode())
+        elif self.key_format == 'u':
+            return(key)
+        elif self.key_format == 'i':
+            return(key)
+        else:
+            print(f"Invalid key format supplied: {self.key_format}")
+            self.usage()
+        
+    def get_key(self, insert):
+        key_struct = insert['u']['key']
+        key = gdb.selected_inferior().read_memory(int(insert) + key_struct['offset'], key_struct['size']).tobytes()
+        decoded_key = self.decode_key(key)
+        return decoded_key
+ 
+    def walk_level(self, head, level, output_file):
+        current = head[0]
+        while current != 0x0:
+            key = self.get_key(current)
+            next = []
+            for i in range(0, 10):
+                next.append(str(current['next'][i]))
+                if (str(current['next'][i]) == "0x0"):
+                    break
+            self.inserts.update({key : insert(key, str(current), next)})
+            self.inserts[key].print(output_file)
+            current = current['next'][0]
+ 
+    def invoke(self, args, from_tty):
+        # Clear the data.
+        self.inserts = {}
+        # Initialize the output file.
+        output_file = open("dump.txt", "w")
```

Comment:
```suggestion
        with open("dump.txt") as output_file:
            # indent all following lines
```
This will auto-close the file on scope end

Reply:
[d83f7e4](https://github.com/wiredtiger/wiredtiger/pull/9406/commits/d83f7e4778537d2e8b29ca16cc502d65a2650293)